nvtabular/ops/operator.py

```diff
@@ -84,7 +84,72 @@ def compute_output_schema(self, input_schema: Schema, col_selector: ColumnSelect
             else output_selector
         )
 
-        return Schema(output_names)
+        # group (old, new) names, grab old if in new columns
+        names_original, names_transformed = [], []
+        for original_column in col_selector.names:
+            # logic works for all ops except TE
+            new_cols = [new_column for new_column in output_names if original_column in new_column]
+            if new_cols:
+                for column in new_cols:
+                    if column not in names_transformed:
+                        names_original.append(original_column)
+                        names_transformed.append(column)
+        # if name is same from input schema means it was a used column,
```

Comment:
It's a small thing, but some whitespace between the sections would make this easier to read and review.
```suggestion

        # if name is same from input schema means it was a used column,
```